pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java

```diff
@@ -189,19 +195,35 @@ public void start() {
 
     public abstract KafkaRecord buildRecord(ConsumerRecord<Object, Object> consumerRecord);
 
+    protected Map<String, String> copyKafkaHeaders(ConsumerRecord<Object, Object> consumerRecord) {
+        Map<String, String> properties = new HashMap<>();
```

Comment:
We can save a allocations by returning Collections.emptyMap here if the feature is not enabled